### PR TITLE
VEGA-285 Browser testing with cypress

### PR DESCRIPTION
**Pull request by hawx**

### CODE REVIEW

File: cypress/integration/teams.spec.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            .each(($el, index) => {
+                cy.wrap($el).should("contain", expected[index]);
+            });
+    });
+
+    it("allows me to search for a team", () => {
+        cy.get("#f-search").clear().type("no such team");
+        cy.get("button[type=submit]").click();
+
+        cy.get(".govuk-table__row").should("have.length", 1);
```
Comment (gregtyler) [inline cypress/integration/teams.spec.js:24]: I know this is just because of pact-mock, but it's confusing that "no such team" would return a single result. Could we change the search to "cool team" so it looks clearer?

File: cypress/support/commands.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// ***********************************************
```
Comment (gregtyler) [inline cypress/support/commands.js:1]: Is this file still needed?

File: cypress/support/index.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+// You can change the location of this file or turn off
+// automatically serving support files with the
+// 'supportFile' configuration option.
+//
+// You can read more here:
+// https://on.cypress.io/configuration
+// ***********************************************************
+
+// Import commands.js using ES2015 syntax:
+import './commands'
```
Comment (gregtyler) [inline cypress/support/index.js:17]: If commands is removed, presumably this can be too

File: docker/pact-stub/pact_stub.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
+package main
```
Comment (gregtyler) [inline docker/pact-stub/pact_stub.go:1]: What differences/benefits does this have over using the official version?

File: cypress/integration/teams.spec.js (hunk 1 above)
Comment (hawx) [reply cypress/integration/teams.spec.js:24]: I think the other confusing thing is the selector I used picks up the header row, hence 1 when there are no results. I'll change to do both.

File: docker/pact-stub/pact_stub.go (hunk 4 above)
Comment (hawx) [reply docker/pact-stub/pact_stub.go:1]: It deals with cookies better. The "official" one compares the cookie headers as strings, whereas mine will match them on key-value, so is immune to Cypress adding it's own weird cookie or reordering the ones you set up.

Comment (gregtyler) [reply docker/pact-stub/pact_stub.go:1]: Cool, I'm not particularly 'gized about supporting our own mock server, but Pact has always been a game of trade-offs.

File: .circleci/config.yml [hunk 5]
```diff
@@ -8,8 +8,10 @@ workflows:
       - lint
       - acceptance-test:
           requires: [test, lint]
-      - push:
+      - cypress:
           requires: [acceptance-test]
+      - push:
+          requires: [cypress]
```
Comment (gregtyler) [inline .circleci/config.yml:14]: CircleCI would be a little bit faster running acceptance-test and cypress in parallel:
```suggestion
      - cypress:
          requires: [test, lint]
      - push:
          requires: [acceptance-test, cypress]
```

**Review by gregtyler — APPROVED** [review]
